decidim/decidim
# Standardize CSV import formats and fix private users CSV import with invalid file

#### :tophat: What? Why?
As discussed at #8298, the CSV file formats could use some standardization so that we wouldn't have different formats in different locations of the admin panel.

At #7084 we introduced a standardized API to read the different import file formats. This PR refactors the following imports to use the readers implementation from that PR:
- The participatory space private users import
- The groups verification import

The problem for standardizing the other import functionality e.g. in verifications is that these APIs live now at the `admin` module. It should be considered as a separate issue to move these under the `core` module to also utilize them e.g. at verifications.

I also left the census import at the `elections` module untouched because it had some additional functionality I did not want to touch in this PR for reading the CSV. But I standardized that module as well as verifications to also use the same configured CSV column separator as the rest of the imports.

Another note to make is that while the import API supports also `.xlsx` and `.json` formats, I did not enable them for these two imports because that would also require some other changes e.g. in the UI examples, etc. which would increase the complexity. This could be also handled as a separate issue (i.e. changing the allowed content types at the import forms to include all possible formats).

This also fixes the actual issue that started the whole discussion at #8298.

#### :pushpin: Related Issues
- Fixes #8298

#### Testing
- Go to import participatory space private participants
- Create a CSV with the provided example but change the column separator to comma `,` (which is invalid)
- Try to submit the form with that file
- Expect to see a flash error instead of an exception
